src/index.ts

```diff
@@ -378,10 +378,14 @@ class QueryPipeline extends EventEmitter {
         if (Object.keys(activePins).length > 0) {
           Object.entries(activePins).forEach(
             ([pinKeyFieldName, pinKeyFieldValues]) => {
-              const fieldValue = valueFromProto(
-                values[pinKeyFieldName]
-              ) as string;
-              if (pinKeyFieldValues.has(fieldValue)) {
+              const fieldValue =
+                (values[pinKeyFieldName] &&
```

Comment:
I think we should probably log to the console when there is an active promotion but the pinKeyFieldName is not included in the requested fields list.

The message should either be ambiguous about _why_ or we can attempt to enumerate likely reasons that this has happened. The set of reasons that I know are:
- The customer has changed their pipeline's default fields to not include `${pinKeyFieldName}` (most likely `_id`).
- The customer has requested a specific set of fields, and `${pinKeyFieldName}` is not in that list.